catboost/libs/metrics/metric.cpp

```diff
@@ -3166,6 +3166,122 @@ void TAverageGain::GetBestValue(EMetricBestValue* valueType, float*) const {
     *valueType = EMetricBestValue::Max;
 }
 
+/* NormalizedGini */
+
+THolder<TNormalizedGiniMetric> TNormalizedGiniMetric::CreateBinClassMetric(double border) {
+    return MakeHolder<TNormalizedGiniMetric>(border);
+}
+
+THolder<TNormalizedGiniMetric> TNormalizedGiniMetric::CreateMultiClassMetric(int positiveClass) {
+    CB_ENSURE(positiveClass >= 0, "Class id should not be negative");
+
+    auto metric = new TNormalizedGiniMetric();
+    metric->PositiveClass = positiveClass;
+    metric->IsMultiClass = true;
+    return metric;
+}
+
+TMetricHolder TNormalizedGiniMetric::Eval(
+        const TVector<TVector<double>>& approx,
+        TConstArrayRef<float> target,
+        TConstArrayRef<float> weightIn,
+        TConstArrayRef<TQueryInfo> /*queriesInfo*/,
+        int begin,
+        int end,
+        NPar::TLocalExecutor& /* executor */
+) const {
+    Y_ASSERT((approx.size() > 1) == IsMultiClass);
+    const auto& approxVec = approx.ysize() == 1 ? approx.front() : approx[PositiveClass];
+    Y_ASSERT(approxVec.size() == target.size());
+
+    TVector<double> targetCopy(target.begin() + begin, target.begin() + end);
+
+    if (!IsMultiClass) {
+        for (ui32 target : targetCopy) {
+            target = target > Border;
```

Comment:
you are modifying local copy, this is noop and probably a bug that need a test case